packages/fgtclb/academic-persons-edit/Classes/Property/TypeConverter/ProfileImageUploadConverter.php

```diff
@@ -237,15 +237,14 @@ private function handleNoFileUploaded(array $uploadedFileInformation): ?ExtbaseF
         }
         return null;
     }
+    */
 
-    private function createFileReferenceFromFalFileReferenceObject(
-        CoreFileReference $falFileReference,
-        ?int $resourcePointer = null
-    ): ExtbaseFileReference {
-        if ($resourcePointer !== null) {
+    private function createFileReferenceFromFalFileReferenceObject(CoreFileReference $falFileReference): ExtbaseFileReference
+    {
+        if ($falFileReference->getIdentifier() !== null) {
             try {
                 // Delete the current profile image with its file reference.
-                $this->resourceFactory->getFileReferenceObject($resourcePointer)->getOriginalFile()->delete();
+                $this->resourceFactory->getFileReferenceObject($falFileReference->getUid())->getOriginalFile()->delete();
```

Comment:
`$falFilReference` is already a core filereference, why do we still retrieve it again instead of directly deleting it / the original file ?

```suggestion
                $falFileReference->getOriginalFile()->delete();
```